fmriprep/viz/reports.py

```diff
@@ -154,20 +154,46 @@ def index_error_dir(self, error_dir):
                         else:
                             scope.set_extra(k, v)
                     scope.level = 'fatal'
-                    message = '%s: %s\n\n' % (node_name, gist)
-                    # 8192 is the message size limit - the important info is probably at the end
-                    message += exception_text[-(8192-len(message)):]
 
-                    # remove file paths
-                    fingerprint = re.sub(r"(/[^/ ]*)+/?", '', message)
-                    # remove words containing numbers
-                    fingerprint = re.sub(r"([a-zA-Z]*[0-9]+[a-zA-Z]*)+", '', fingerprint)
-                    # adding the return code if it exists
-                    for line in message.split('\n'):
-                        if line.startswith("Return code"):
-                            fingerprint += line
+                    # Group common events with pre specified fingerprints
+                    fingerprint_dict = {'permission-denied': ["PermissionError: [Errno 13] "
+                                                              "Permission denied"],
+                                        'memory-error': ["MemoryError", "Cannot allocate memory"],
+                                        'reconall-already-running': ["ERROR: it appears that "
+                                                                     "recon-all is already "
+                                                                     "running"],
+                                        'no-disk-space': ["OSError: [Errno 28] No space left on "
+                                                          "device", "[Errno 122] Disk quota "
+                                                                    "exceeded"],
+                                        'sigkill': ["Return code: 137"],
+                                        'keyboard-interrupt': ["KeyboardInterrupt"]}
+
+                    fingerprint = ''
+                    issue_title = node_name + ': ' + gist
+                    for new_fingerprint, error_snippets in fingerprint_dict.items():
+                        for error_snippet in error_snippets:
+                            if error_snippet in crash_info['traceback']:
+                                fingerprint = new_fingerprint
+                                issue_title = new_fingerprint
+                                break
+                        if fingerprint:
                             break
 
+                    message = issue_title + '\n\n'
+                    message += exception_text[-(8192-len(message)):]
+                    if fingerprint:
+                        self.sentry_sdk.add_breadcrumb(fingerprint, 'fatal')
+                    else:
```

Comment:
is it intended that fatal fingerprints do not seem to be stripped off of paths and words containing numbers?

Reply:
fatal fingerprints consist only of a single string (such as `sigkill`) and therefore  do not include any paths or numbers